tests/python/contrib/test_hexagon/topi/test_conv2d_fp16_intrin.py

```diff
@@ -0,0 +1,256 @@
+            target=target,
+            name=func_name,
+        )
+
+    return module
+
+
+shape_parameters = [
+    {
+        "act_shape": (1, 8, 4, 3),
+        "wgt_shape": (3, 3, 3, 3),
+        "inp_offset": (0, 0),
+        "inp_stride": (1, 1),
+    },
+    {
+        "act_shape": (1, 10, 14, 3),
+        "wgt_shape": (3, 3, 3, 3),
+        "inp_offset": (0, 0),
+        "inp_stride": (1, 1),
+    },
+    {
+        "act_shape": (1, 14, 6, 3),
+        "wgt_shape": (3, 3, 3, 3),
+        "inp_offset": (0, 0),
+        "inp_stride": (1, 1),
+    },
+    {
+        "act_shape": (1, 14, 6, 3),
+        "wgt_shape": (3, 3, 3, 64),
+        "inp_offset": (0, 0),
+        "inp_stride": (1, 1),
+    },
+    {
+        "act_shape": (1, 14, 6, 3),
+        "wgt_shape": (5, 5, 3, 3),
+        "inp_offset": (0, 0),
+        "inp_stride": (1, 1),
+    },
+    {
+        "act_shape": (1, 8, 8, 3),
+        "wgt_shape": (2, 2, 3, 3),
+        "inp_offset": (0, 0),
+        "inp_stride": (1, 1),
+    },
+    {
+        "act_shape": (1, 14, 6, 64),
+        "wgt_shape": (3, 3, 64, 3),
+        "inp_offset": (0, 0),
+        "inp_stride": (1, 1),
+    },
+    {
+        "act_shape": (1, 4, 4, 40),
+        "wgt_shape": (3, 3, 40, 3),
+        "inp_offset": (0, 0),
+        "inp_stride": (1, 1),
+    },
+    {
+        "act_shape": (1, 4, 4, 3),
+        "wgt_shape": (3, 3, 3, 3),
+        "inp_offset": (0, 0),
+        "inp_stride": (1, 1),
+    },
+    {
+        "act_shape": (1, 5, 5, 3),
+        "wgt_shape": (3, 3, 3, 3),
+        "inp_offset": (0, 0),
+        "inp_stride": (1, 1),
+    },
+    {
+        "act_shape": (1, 6, 6, 3),
+        "wgt_shape": (3, 3, 3, 3),
+        "inp_offset": (0, 0),
+        "inp_stride": (1, 1),
+    },
+    {
+        "act_shape": (1, 7, 7, 3),
+        "wgt_shape": (3, 3, 3, 3),
+        "inp_offset": (0, 0),
+        "inp_stride": (1, 1),
+    },
+    {
+        "act_shape": (1, 8, 8, 3),
+        "wgt_shape": (3, 3, 3, 3),
+        "inp_offset": (0, 0),
+        "inp_stride": (1, 1),
+    },
+    {
+        "act_shape": (1, 8, 8, 3),
+        "wgt_shape": (5, 5, 3, 3),
+        "inp_offset": (0, 0),
+        "inp_stride": (1, 1),
+    },
+    {
+        "act_shape": (1, 8, 8, 64),
+        "wgt_shape": (2, 2, 64, 64),
+        "inp_offset": (0, 0),
+        "inp_stride": (1, 1),
+    },
+    {
+        "act_shape": (1, 8, 4, 3),
+        "wgt_shape": (3, 3, 3, 3),
+        "inp_offset": (0, 0),
+        "inp_stride": (2, 2),
+    },
+    {
+        "act_shape": (1, 14, 6, 3),
+        "wgt_shape": (3, 3, 3, 64),
+        "inp_offset": (0, 0),
+        "inp_stride": (2, 2),
+    },
+    {
+        "act_shape": (1, 14, 6, 3),
+        "wgt_shape": (5, 5, 3, 3),
+        "inp_offset": (0, 0),
+        "inp_stride": (2, 2),
+    },
+    {
+        "act_shape": (1, 8, 8, 3),
+        "wgt_shape": (2, 2, 3, 3),
+        "inp_offset": (0, 0),
+        "inp_stride": (2, 2),
+    },
```

Comment:
These are all very small convolutions. Can we add some that are representative of real workloads and mark them with the `slow` decorator, or skip.

Reply:
I tried adding larger convolution tests earlier, but realized that it won't work due to VTCM limit. I can still add them and mark them as skip if you think that's a good idea for the future, but as it stands now, the op has to be split into chunks to accommodate VTCM size.